charmhelpers/contrib/openstack/vaultlocker.py

```diff
@@ -52,29 +71,70 @@ def __call__(self):
                     token = json.loads(token)
                     vault_url = json.loads(vault_url)
 
+                    if token not in current_tokens:
+                        if "sentinel" in current_tokens:
+                            current_tokens = [token]
+                        else:
+                            current_tokens.append(token)
+
+                    # If we have already had a successful request then all we
+                    # are doing is collecting current tokens so that we can
+                    # purge old ones.
+                    if self.complete:
```

Comment:
Superclass (OSContextGenerator) already has an attribute called "complete", that's why this is not currently failing for not being initialized. Please change this attribute name and initialize it.

Reply:
Why do you expect this to fail? That variable is initialised to false by the base class which is what I expect so  don't think there is an issue here.